docs/sphinx/source/introexamples.rst

```diff
@@ -0,0 +1,246 @@
+.. _introexamples:
+
+Intro Examples
+==============
+
+This page contains introductory examples of pvlib python usage.
+
+.. _modeling-paradigms:
+
+Modeling paradigms
+------------------
+
+The backbone of pvlib-python
+is well-tested procedural code that implements PV system models.
+pvlib-python also provides a collection of classes for users
+that prefer object-oriented programming.
+These classes can help users keep track of data in a more organized way,
+provide some "smart" functions with more flexible inputs,
+and simplify the modeling process for common situations.
+The classes do not add any algorithms beyond what's available
+in the procedural code, and most of the object methods
+are simple wrappers around the corresponding procedural code.
+
+Let's use each of these pvlib modeling paradigms
+to calculate the yearly energy yield for a given hardware
+configuration at a handful of sites listed below.
+
+.. ipython:: python
+
+    import pandas as pd
+    import matplotlib.pyplot as plt
+
+    naive_times = pd.DatetimeIndex(start='2015', end='2016', freq='1h')
+
+    # very approximate
+    # latitude, longitude, name, altitude, timezone
+    coordinates = [(30, -110, 'Tucson', 700, 'Etc/GMT+7'),
+                   (35, -105, 'Albuquerque', 1500, 'Etc/GMT+7'),
+                   (40, -120, 'San Francisco', 10, 'Etc/GMT+8'),
+                   (50, 10, 'Berlin', 34, 'Etc/GMT-1')]
+
+    import pvlib
+
+    # get the module and inverter specifications from SAM
+    sandia_modules = pvlib.pvsystem.retrieve_sam('SandiaMod')
+    sapm_inverters = pvlib.pvsystem.retrieve_sam('cecinverter')
+    module = sandia_modules['Canadian_Solar_CS5P_220M___2009_']
+    inverter = sapm_inverters['ABB__MICRO_0_25_I_OUTD_US_208_208V__CEC_2014_']
+
+    # specify constant ambient air temp and wind for simplicity
+    temp_air = 20
+    wind_speed = 0
+
+
+Procedural
+^^^^^^^^^^
+
+The straightforward procedural code can be used for all modeling
+steps in pvlib-python.
+
+The following code demonstrates how to use the procedural code
+to accomplish our system modeling goal:
+
+.. ipython:: python
+
+    system = {'module': module, 'inverter': inverter,
+              'surface_azimuth': 180}
+
+    energies = {}
+
+    for latitude, longitude, name, altitude, timezone in coordinates:
+        times = naive_times.tz_localize(timezone)
+        system['surface_tilt'] = latitude
+        solpos = pvlib.solarposition.get_solarposition(times, latitude, longitude)
+        dni_extra = pvlib.irradiance.get_extra_radiation(times)
+        dni_extra = pd.Series(dni_extra, index=times)
```

Comment:
Is this explicit conversion to `pd.Series` still needed?

Reply:
no, good catch